clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -206,9 +206,9 @@ public synchronized void unsubscribe() {
                         subscriptions.position(entry.getKey(), newPosition);
                     }
                 }
+                entry.getValue().clear();
```

Comment:
Found this bug while working on the unit tests: when a partition is paused, we should not clear its records since they would not be returned; we should only clear the corresponding records which are included in the results.

Reply:
@guozhangwang There is already a PR for this issue (https://github.com/apache/kafka/pull/7505) Can you maybe review and merge this PR? Thanks to @ableegoldman for pointing out the overlap.

Reply:
Done.